purchase_reception_status_line/models/purchase_order_line.py

```diff
@@ -6,46 +6,40 @@
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = "purchase.order.line"
+    _name = "purchase.order.line"
+    _inherit = ["purchase.order.line", "mail.thread", "mail.activity.mixin"]
 
     reception_status = fields.Selection(
         [
             ("no", "Nothing Received"),
             ("partial", "Partially Received"),
             ("received", "Fully Received"),
+            ("over", "Over Received"),
         ],
         compute="_compute_reception_status",
         store=True,
     )
     force_received = fields.Boolean(
+        string="Force closed by Buyer",
         readonly=False,
         states={"draft": [("readonly", True)]},
-        compute="_compute_force_received",
         store=True,
         copy=False,
         help="If true, the reception status will be forced to Fully Received, "
         "even if some quantities are not fully received. ",
+        tracking=True,
     )
 
-    @api.depends("order_id.force_received")
-    def _compute_force_received(self):
-        prec = self.env["decimal.precision"].precision_get("Product Unit of Measure")
+    def action_commute_force_received(self):
         for rec in self:
-            if (
-                rec.order_id.force_received
-                and not float_compare(
-                    rec.qty_received, rec.product_qty, precision_digits=prec
-                )
-                >= 0
-            ):
-                rec.force_received = True
+            rec.force_received = not rec.force_received
+            rec._compute_reception_status()
```

Comment:
The call for `_compute_reception_status` shouldn't be needed, right? `force_received` is already in the `depends`